pkg/vault/test/foundry/BatchRouterERC4626Pool.t.sol

```diff
@@ -0,0 +1,409 @@
+
+        _prankStaticCall();
+        uint256 expectBPTOut = router.queryAddLiquidityUnbalanced(erc4626Pool, exactUnderlyingAmountsIn, new bytes(0));
+
+        uint256 beforeUSDCBalance = usdc.balanceOf(alice);
+        uint256 beforeDAIBalance = dai.balanceOf(alice);
+        (uint256 beforeWaUSDCBufferBalanceUnderlying, uint256 beforeWaUSDCBufferBalanceWrapped) = vault
+            .getBufferBalance(waUSDC);
+        (uint256 beforeWaDAIBufferBalanceUnderlying, uint256 beforeWaDAIBufferBalanceWrapped) = vault.getBufferBalance(
+            waDAI
+        );
+
+        vm.prank(alice);
+        uint256 bptOut = batchRouter.addLiquidityUnbalancedToERC4626Pool(
+            erc4626Pool,
+            exactUnderlyingAmountsIn,
+            1,
+            false,
+            new bytes(0)
+        );
+
+        {
+            uint256 afterUSDCBalance = usdc.balanceOf(alice);
+            assertEq(beforeUSDCBalance - afterUSDCBalance, operationAmount, "USDC balance should decrease");
+        }
+        {
+            uint256 afterDAIBalance = dai.balanceOf(alice);
+            assertEq(beforeDAIBalance - afterDAIBalance, operationAmount, "DAI balance should decrease");
+        }
+        {
+            (uint256 afterWaUSDCBufferBalanceUnderlying, uint256 afterWaUSDCBufferBalanceWrapped) = vault
+                .getBufferBalance(waUSDC);
+            assertEq(
+                beforeWaUSDCBufferBalanceWrapped - afterWaUSDCBufferBalanceWrapped,
+                operationAmount,
+                "waUSDC wrapped buffer balance should decrease"
+            );
+            assertEq(
+                afterWaUSDCBufferBalanceUnderlying - beforeWaUSDCBufferBalanceUnderlying,
+                operationAmount,
+                "waUSDC underlying buffer balance should increase"
+            );
+        }
+        {
+            (uint256 afterWaDAIBufferBalanceUnderlying, uint256 afterWaDAIBufferBalanceWrapped) = vault
+                .getBufferBalance(waDAI);
+            assertEq(
+                beforeWaDAIBufferBalanceWrapped - afterWaDAIBufferBalanceWrapped,
+                operationAmount,
+                "waDAI wrapped buffer balance should decrease"
+            );
+            assertEq(
+                afterWaDAIBufferBalanceUnderlying - beforeWaDAIBufferBalanceUnderlying,
+                operationAmount,
+                "waDAI underlying buffer balance should increase"
+            );
+        }
+
+        assertApproxEqAbs(bptOut, expectBPTOut, DELTA, "BPT operationAmount should match expected");
+        assertEq(IERC20(address(erc4626Pool)).balanceOf(alice), bptOut, "BPT balance should increase");
+    }
+
+    function testAddLiquidityUnbalancedToERC4626PoolWhenStaticCall() public checkBuffersWhenStaticCall(alice) {
+        uint256[] memory exactUnderlyingAmountsIn = [operationAmount, operationAmount].toMemoryArray();
+
+        vm.prank(alice, address(0));
```

Comment:
_prankStaticCall()?

Reply:
`_prankStaticCall` also mocks the sender to zero address but we need the sender because vault checks tokens from the sender address

Reply:
Hm, maybe I'm mistaken

Reply:
yes, I'm right